stage/src/stage.rs

```diff
@@ -65,36 +65,33 @@ impl Stage {
     }
 
     pub fn dispatch(&mut self) {
-        if self.split_task.state == TASK_STATE_INITIAL {
-            self.gen_split_task();
-            return;
-        }
-        if self.split_task.state == TASK_STATE_SUCCESS {
-            if self.prove_tasks.is_empty() {
-                self.gen_prove_task();
-                return;
-            }
-        } else {
-            return;
-        }
-        let mut all_prove_task_success = true;
-        for prove_task in &self.prove_tasks {
-            if prove_task.state != TASK_STATE_SUCCESS {
-                all_prove_task_success = false;
-                break;
+        match self.split_task.state {
+            TASK_STATE_INITIAL => self.gen_split_task(),
+            TASK_STATE_SUCCESS => {
+                if self.prove_tasks.is_empty() {
+                    self.gen_prove_task();
+                } else {
+                    let mut all_prove_task_success = true;
+                    for prove_task in &self.prove_tasks {
```

Comment:
we can use  iterator::all to check if all tasks succeed.  https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.all